src/testrender/shading.cpp

```diff
@@ -624,115 +722,179 @@ struct Reflection final : public BSDF, ReflectionParams {
+                    cw, *comp->as<OrenNayarParams>());
+                break;
+            case TRANSLUCENT_ID:
+                ok = result.bsdf.add_bsdf<Diffuse<1>, DiffuseParams>(
+                    cw, *comp->as<DiffuseParams>());
+                break;
+            case PHONG_ID:
+                ok = result.bsdf.add_bsdf<Phong, PhongParams>(
+                    cw, *comp->as<PhongParams>());
+                break;
+            case WARD_ID:
+                ok = result.bsdf.add_bsdf<Ward, WardParams>(
+                    cw, *comp->as<WardParams>());
+                break;
+            case MICROFACET_ID: {
+                const MicrofacetParams* mp = comp->as<MicrofacetParams>();
+                if (mp->dist == u_ggx) {
+                    switch (mp->refract) {
+                    case 0:
+                        ok = result.bsdf.add_bsdf<MicrofacetGGXRefl,
+                                                  MicrofacetParams>(cw, *mp);
+                        break;
+                    case 1:
+                        ok = result.bsdf.add_bsdf<MicrofacetGGXRefr,
+                                                  MicrofacetParams>(cw, *mp);
+                        break;
+                    case 2:
+                        ok = result.bsdf.add_bsdf<MicrofacetGGXBoth,
+                                                  MicrofacetParams>(cw, *mp);
+                        break;
+                    }
+                } else if (mp->dist == u_beckmann || mp->dist == u_default) {
+                    switch (mp->refract) {
+                    case 0:
+                        ok = result.bsdf.add_bsdf<MicrofacetBeckmannRefl,
+                                                  MicrofacetParams>(cw, *mp);
+                        break;
+                    case 1:
+                        ok = result.bsdf.add_bsdf<MicrofacetBeckmannRefr,
+                                                  MicrofacetParams>(cw, *mp);
+                        break;
+                    case 2:
+                        ok = result.bsdf.add_bsdf<MicrofacetBeckmannBoth,
+                                                  MicrofacetParams>(cw, *mp);
+                        break;
+                    }
+                }
+                break;
+            }
+            case REFLECTION_ID:
+            case FRESNEL_REFLECTION_ID:
+                ok = result.bsdf.add_bsdf<Reflection, ReflectionParams>(
+                    cw, *comp->as<ReflectionParams>());
+                break;
+            case REFRACTION_ID:
+                ok = result.bsdf.add_bsdf<Refraction, RefractionParams>(
+                    cw, *comp->as<RefractionParams>());
+                break;
+            case TRANSPARENT_ID:
+                ok = result.bsdf.add_bsdf<Transparent, int>(cw, 0);
+                break;
+            }
+            OSL_ASSERT(ok && "Invalid closure invoked in surface shader");
+        }
+        break;
+    }
+    }
 }
```

Comment:
This was more readable prior to clang-format.

Reply:
about to get rewritten anyway